packages/main/src/Form.ts

```diff
@@ -0,0 +1,456 @@
+ *
+ * ### Items Label Span
+ *
+ * The placement of the labels depends on the size of the used column.
+ * If there is enough space, the labels are next to their associated fields, otherwise  - above the fields.
+ * By default, the labels take 4/12 of the FormItem, leaving 8/12 parts to associated fields.
+ * You can control what space the labels should take via the `labelSpan` property.
+ *
+ * **For example:** To always place the labels on top set: `labelSpan="S12 M12 L12 XL12"` property.
+ *
+ * ### ES6 Module Import
+ *
+ * - import @ui5/webcomponents/dist/Form.js";
+ * - import @ui5/webcomponents/dist/FormGroup.js";
+ * - import @ui5/webcomponents/dist/FormItem.js";
+ *
+ * @csspart header - Used to style the wrapper of the header.
+ * @csspart layout - Used to style the element defining the form column layout.
+ * @csspart column - Used to style a single column of the form column layout.
+ *
+ * @public
+ * @since 2.0.0
+ */
+@customElement({
+	tag: "ui5-form",
+	renderer: litRender,
+	styles: FormCss,
+	template: FormTemplate,
+	dependencies: [Title],
+})
+class Form extends UI5Element {
+	/**
+	 * Defines the number of columns to distribute the form content by breakpoint.
+	 *
+	 * Supported values:
+	 * - `S` - 1 column by default (1 column is recommended)
+	 * - `M` - 1 column by default (up to 2 columns are recommended)
+	 * - `L` - 2 columns by default (up to 3 columns are recommended)
+	 * - `XL` - 2 columns by default (up to 6 columns  are recommended)
+	 *
+	 * @default "S1 M1 L2 XL2"
+	 * @public
+	 */
+	@property({ type: String, defaultValue: "S1 M1 L2 XL2" })
+	layout!: string;
+
+	/**
+	 * Defines the width proportion of the labels and fields of a FormItem by breakpoint.
+	 *
+	 * By default, the labels take 4/12 (or 1/3) of the form item in M,L and XL sizes,
+	 * and 12/12 in S size, e.g in S the label is on top of its associated field.
+	 *
+	 * The supported values are between 1 and 12. Greater the number, more space the label will use.
+	 *
+	 * **Note:** If "12" is set, the label will be displayed on top of its assosiated field.
+	 * @default "S12 M4 L4 XL4"
+	 * @public
+	 */
+	@property({ type: String, defaultValue: "S12 M4 L4 XL4" })
+	labelSpan!: string;
+
+	/**
+	 * Defines the header text of the component.
+	 *
+	 * **Note:** The property gets overridden by the `header` slot.
+	 *
+	 * @default ""
+	 * @public
+	 */
+	@property()
+	headerText!: string;
+
+	/**
+	 * Defines the vertical spacing between form items.
+	 *
+	 * **Note:** If the Form is meant to be switched between "non-edit" and "edit" modes,
+	 * we recommend using "Large" item spacing in "non-edit" mode, and "Normal" - for "edit" mode,
+	 * to avoid "jumping" effect, caused by the hight difference between texts in "non-edit" mode and the input fields in "edit" mode.
+	 *
+	 * @default "Normal"
+	 * @public
+	 */
+	@property({ type: FormItemSpacing, defaultValue: FormItemSpacing.Normal })
+	itemSpacing!: FormItemSpacing;
```

Comment:
minor: use template literals for properties of type enum. also applying for form item and form group item.